File: src/com/douma/_10_day_栈和队列/单调栈/RightFirstLarger.js
var findRightSmall = function(nums) {
    const ans = new Array(nums.length).fill(-1)
    const stack = []
    for (let i = 0; i < nums.length; i++) {
        const x = nums[i]
        // 单调递减栈
        while (stack.length && x > nums[stack[stack.length - 1]]) {
            ans[stack[stack.length - 1]] = i
            stack.pop()
        }
        stack.push(i)
    }
    return ans
}

File: src/com/douma/_10_day_栈和队列/单调栈/RightFirstLarger.py
def findRightSmall(self, nums: List[int]) -> List[int]:
    ans = [-1] * len(nums)
    stack = []
    for i in range(len(nums)):
        x = nums[i]
        # 单调递减栈
        while not stack and x > nums[stack[-1]]:
            ans[stack[-1]] = i
            stack.pop()
        stack.append(i)
    return ans